frontend/src/lib/rankings.ts

```diff
@@ -29,6 +35,35 @@ export const rankedToRankings = (ranked: RankedSites): SiteRakings => {
   return result;
 };
 
+// Map regexs to their respective rankings
+const reToRankMap = {
+  [Ranking.LIKED]: /Like\(\s*Site\s*\(\s*"(\S*)"\s*\)\s*\);/g,
+  [Ranking.DISLIKED]: /Dislike\(\s*Site\s*\(\s*"(\S*)"\s*\)\s*\);/g,
+  [Ranking.BLOCKED]: /Rule\s*{\s*Matches\s*{\s*Site\("\|(\S*)\|"\),\s*},\s*Action\(Discard\)\s*};/g
+}
+
+export const importedOpticToRankings = (optic: string): SiteRankings => {
+  const rankedSites: RankedSites = {
+    liked: [],
+    disliked: [],
+    blocked: [],
+  };
+
+  for(const [_, ranking] of Object.entries(Ranking)){
+    let re = reToRankMap[ranking]
+    let match: RegExpExecArray | null
+
+    do {
+      match = re.exec(optic);
+      if (match) {
+        rankedSites[ranking].push(match[1])
+      }
+    } while (match)
+  }
+
+  return rankedToRankings(rankedSites)
+}
+
```

Comment:
I think we should keep all optics parsing inside the `optics` crate in case we want to make changes to the syntax later.
The most elegant solution would probably be to create a wasm crate that exposes this functionality to the frontend so the parsing still happens client-side. Alternatively we can create a `/beta/api/hosts/import` endpoint for the time being and then move the functionality to a wasm create later.

Reply:
Totally makes sense. To me WASM makes the most sense to preserve some of your resources on the server side. Been looking for an excuse to play with it for a while anyway. Some optic validation would be nice too, is that the goal with #147?